compiler/rustc_middle/src/hir/map/collector.rs

```diff
@@ -124,53 +102,20 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
         IndexedHir { map: self.map, parenting: self.parenting }
     }
 
-    fn insert_entry(&mut self, id: HirId, entry: Entry<'hir>, hash: Fingerprint) {
-        let i = id.local_id.as_u32() as usize;
-
-        let arena = self.arena;
+    fn insert_owner(&mut self, owner: LocalDefId, node: OwnerNode<'hir>) {
```

Comment:
I don't really understand the logic behind all this restructuring in `compiler/rustc_middle/src/hir/map/collector.rs`.

Reply:
The objective is to separate the owner vs non-owner code paths.